### PR TITLE
added options to log to username

### PR DESCRIPTION
**Pull request by Mattihew**

### CODE REVIEW

File: src/main/java/com/logdyn/api/endpoints/LogSession.java [hunk 1]
```diff
@@ -1,92 +1,117 @@
...
+			return session.getAsyncRemote().sendText(LogRecordUtils.toJSON(messages));
 		}
-		
-		return sessionId;
+		return null;
 	}
-    
-    private static JSONObject logRecordToJSON(final LogRecord logRecord)
+
+	public void addExistingSession(final LogSession logSession)
```
Comment (jsjlewis96) [inline src/main/java/com/logdyn/api/endpoints/LogSession.java]: Javadoc please :)

Comment (jake-lewis) [inline src/main/java/com/logdyn/api/endpoints/LogSession.java]: Javadoc please

File: src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java [hunk 2]
```diff
@@ -22,129 +22,137 @@
...
+		{
+			LoggingEndpoint.ROOT_SESSION.removeWebsocketSession(session);
+			logSession.logMessage(
+					new LogMessage(Level.FINER, "Session closed due to " + closeReason.getReasonPhrase()),
+					session);
+		}
 	}
 	
-	public static void log(final LogRecord logRecord)
+	public static boolean log(final LogRecord logRecord)
```
Comment (jake-lewis) [inline src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java]: javadoc, also I think general comments throughout this class, as it's the most complex stuff and not the easiest thing to read straight off the bat

File: src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java [hunk 3]
```diff
@@ -22,129 +22,137 @@
...
 			httpSessionId = ((LogMessage) logRecord).getSessionId();
+			username = ((LogMessage) logRecord).getUsername();
 		}
 		
-		LoggingEndpoint.getLogSession(username, httpSessionId).logMessage(logRecord);
+		return LoggingEndpoint.getLogSession(username, httpSessionId).logMessage(logRecord);
 	}
 
-	public static void clearSession(final String httpSessionId)
+	public static boolean clearSession(final String httpSessionId)
```
Comment (jake-lewis) [inline src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java]: javadoc

File: src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java [hunk 4]
```diff
@@ -22,129 +22,137 @@
...
 	{
-		LoggingEndpoint.NON_USER_SESSIONS.remove(httpSessionId);
+		return LoggingEndpoint.NON_USER_SESSIONS.remove(httpSessionId) != null;
 	}
-	
-	private static Level parseLevel(final JSONObject jsonObject)
+
+	public static boolean clearUser(final String username)
```
Comment (jake-lewis) [inline src/main/java/com/logdyn/api/endpoints/LoggingEndpoint.java]: javadoc

File: src/main/java/com/logdyn/api/model/LogMessage.java [hunk 5]
```diff
@@ -1,141 +1,141 @@
...
+{
+
+	/** Generated serialID **/
+	private static final long serialVersionUID = -9093297911420796177L;
+	
+	private final String sessionId;
+	private final String username;
+	private int hashCode = -1;
+
+	public LogMessage (final Level level, final String message)
```
Comment (jake-lewis) [inline src/main/java/com/logdyn/api/model/LogMessage.java:24]: ALL the javadoc that has been removed should probably be replaced

**Review by jake-lewis — CHANGES_REQUESTED** [review]

File: src/main/java/com/logdyn/api/model/LogMessage.java (hunk 5 above)
Comment (Mattihew) [reply src/main/java/com/logdyn/api/model/LogMessage.java:24]: this whole class will hopfully be removed soon. i'll comment it if it turns out its staying

**Review by jake-lewis — APPROVED** [review]

**Pull request merged**